src/components/text/text.tsx

```diff
@@ -30,13 +30,13 @@ export type EuiTextProps = CommonProps &
     /**
      * Any of our named colors or a `hex`, `rgb` or `rgba` value.
      */
-    color?: TextColor | CSSProperties['color'];
+    color?: EuiTextColorProps['color'];
```

Comment:
so - I'm going to be honest, I'm not super sold on this PR necessarily being the best approach. It adds a **lot** of extra cruft (extra class/color overrides, etc) and negates almost the entire point of the `euiText()` mixin (which I think is mostly valid; we default to either inherit or text color and we don't need to re-override it via another text color).

If all we want is the props table to show a default value, this is all we need to do:

```suggestion
    /**
     * Any of our named colors or a `hex`, `rgb` or `rgba` value.
     * @default inherit
     */
    color?: TextColor | CSSProperties['color'];
```

Reply:
It was not adding a color override. Just the extra class `euiTextColor-inherit`:

<img width="465" alt="Screenshot 2022-09-28 at 19 19 58" src="https://user-images.githubusercontent.com/2750668/192859045-bf7c4bf5-7022-4e00-9e00-66b6feebbfd7.png">

The idea for this PR comes from a question in our slack channel. That made me realize that is not easy to see what is the current color being applied to **EuiText**. You need to inspect and look into the styles to realize that it's getting `inherit`. No CSS class name, no types showing that there is an `inherit` option or what is the default. 
 
<img width="314" alt="Screenshot 2022-09-28 at 19 19 47" src="https://user-images.githubusercontent.com/2750668/192860638-5db4eaeb-eb18-4a81-97ba-3b474b627067.png">


If `inherit` is the default, the component should assume it. For me, having the extra class `euiTextColor-inherit` doesn't bother me... 

But the `@default inherit` that @constancecchen suggested helps with what I was trying to achieve. So I pushed [df9f0f1](https://github.com/elastic/eui/pull/6267/commits/df9f0f1af6e3dab4874e729c133c46f157322e0b) reverting some changes and adding `@default inherit`.

Reply:
Ahhh gotcha, I assumed that EuiText was setting a `color: inherit/text` and then EuiTextColor was coming in later and setting `color: inherit` on top of that. Either way though, I lean towards thinking the core Slack question is satisfied by updating our props table via `@default` and I like that this is a lower touch approach with less side effects.